src/components/generator/EditDialog.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import FetchHandler from "../common/FetchHandler";
+import Settings from "../common/settings";
+import Notifications from "../common/Notifications";
+import Editor from "../editor/Editor";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, IconButton, styled } from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import { Link, useNavigate, useParams } from "react-router-dom";
+
+const EditDialogRoot = styled('div')({
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+
+  '> *': {
+    marginBottom: '16px',
+  }
+});
+
+const StyledButton = styled(Button)({
+  cursor: 'pointer',
+  width: '128px',
+
+  ':hover': {
+    backgroundColor: '#BFBFBF',
+  },
+});
+
+const StyledIconButton = styled(IconButton)({
+    position: 'absolute',
+    top: 10,
+    right: 0,
+  });
```

Comment:
See comment on `StyledIconButton` in `GenerateDialog.js`

Reply:
implemented in branch: "generatorpage-add-delete"